pyccl/nl_pt/ept.py

```diff
@@ -0,0 +1,684 @@
+import numpy as np
+from ..pk2d import Pk2D
+from ..base import CCLAutoreprObject, unlock_instance
+
+
+class EulerianPTCalculator(CCLAutoreprObject):
+    """ This class implements a set of methods that can be
+    used to compute the various components needed to estimate
+    Eulerian perturbation theory correlations. These calculations
+    are currently based on FAST-PT
+    (https://github.com/JoeMcEwen/FAST-PT).
+
+    In the parametrisation used here, the galaxy overdensity
+    is expanded as:
+
+    .. math::
+        \\delta_g=b_1\\,\\delta+\\frac{b_2}{2}\\delta^2+
+        \\frac{b_s}{2}s^2+\\frac{b_{3nl}}{2}\\psi_{nl}+
+        \\frac{b_{k2}}{2}\\nabla^2\\delta.
+
+    In turn, the intrinsic alignment component is expanded as
+
+    .. math::
+        s^I_{ij}=c_1\\,s_{ij}+c_2(s_{ik}s_{jk}-s^2\\delta_{ik}/3)
+        +c_\\delta\\,\\delta\\,s_{ij}
+
+    (note that the higher-order terms are not divided by 2!).
+
+    .. note:: Only the leading-order non-local term (i.e.
+              :math:`\\langle \\delta\\,\\nabla^2\\delta`) is
+              taken into account in the expansion. All others are
+              set to zero.
+
+    .. note:: Terms of the form
+              :math:`\\langle \\delta^2 \\psi_{nl}\\rangle` (and
+              likewise for :math:`s^2`) are set to zero.
+
+    .. note:: The full non-linear model for the cross-correlation
+              between number counts and intrinsic alignments is
+              still work in progress in FastPT. As a workaround
+              CCL assumes a non-linear treatment of IAs, but only
+              linearly biased number counts.
```

Comment:
Can we have a warning that gets generated when trying to compute P_gi with a non-linear galaxy tracer? I'm concerned that this note in the documentation might get missed.

Reply:
Done